qutebrowser/browser/webkit/webkithistory.py

```diff
@@ -48,7 +50,8 @@ def historyContains(self, url_string):
         Return:
             True if the url is in the history, False otherwise.
         """
-        return url_string in self._history.history_dict
+        with debug.log_time('sql', 'historyContains'):
+            return url_string in self._history
```

Comment:
Looks like `lru_cache` was indeed more appropriate here, but mainly because it can cache both `True` and `False` with a correct cache clear (when the history changed).

I did so in 939d2823ed3c191303d04a965cc43e3728ab819f and it went down from 16000 calls to `historyContains` (why?!?) when loading heise.de to something like 250...